Implement Redis 7.0.0 [P]EXPIRE[AT] options

michael-grunder opened the pull request:
See #2068

michael-grunder commented:
@yatsukhnenko I wasn't really sure the best way to handle the new option.

This way is flexible, but I wonder if we should instead take an options array for future-proofing (e.g. if Redis adds more exotic options to this command).

yatsukhnenko commented:
@michael-grunder currently in `set` and `zadd` commands we use array to handle NX/EX/etc options so maybe it worth to implement it here but I'm totally fine with current implementation.

yatsukhnenko commented on redis_commands.c:
```diff
@@ -5724,6 +5724,32 @@ int redis_xtrim_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
+        Z_PARAM_STR(key)
+        Z_PARAM_LONG(timeout)
+        Z_PARAM_OPTIONAL
+        Z_PARAM_STR(mode)
```
Consider to add check for `mode` values

michael-grunder committed:
Implement Redis 7.0.0 [P]EXPIRE[AT] options

See #2068

michael-grunder commented:
@yatsukhnenko I added a check on the mode argument and updated our stubs/docs with actual documentation.

I think maybe we should break this command in PhpRedis 6.0.0 to instead return an integer (`0` when an expiration was not set, `1` when an expiration was set, and `false` on failure).
